src/components/calcite-block-section/calcite-block-section.tsx

```diff
@@ -68,28 +79,44 @@ export class CalciteBlockSection {
   //
   // --------------------------------------------------------------------------
 
-  onHeaderClick = () => {
+  toggleSection = () => {
     this.open = !this.open;
     this.calciteBlockSectionToggle.emit();
   };
 
+  handleSwitchClick = ({ currentTarget, target }: MouseEvent) => {
+    const isDupeClick = target === currentTarget;
+
+    if (isDupeClick) {
+      return;
+    }
+
+    this.toggleSection();
+  };
+
   // --------------------------------------------------------------------------
   //
   //  Render Methods
   //
   // --------------------------------------------------------------------------
 
   render() {
-    const { el, open, textCollapse, textExpand } = this;
+    const { el, open, text, textCollapse, textExpand, toggle } = this;
     const dir = getElementDir(el);
     const arrowIcon = open ? caretDown16F : dir === "rtl" ? caretLeft16F : caretRight16F;
     const toggleLabel = open ? textCollapse : textExpand;
 
-    const headerNode = (
+    const headerNode = toggle ? (
+      <label aria-label={toggleLabel} class={classnames(CSS.toggle, CSS.toggleSwitch)}>
+        {text}
+        <calcite-switch switched={open} onClick={this.handleSwitchClick} />
+      </label>
+    ) : (
       <calcite-action
         aria-label={toggleLabel}
-        onClick={this.onHeaderClick}
-        text={this.text}
+        class={CSS.toggle}
+        onClick={this.toggleSection}
+        text={text}
```

Comment:
question: can you just do `text` here?

Reply:
I don't follow. You mean as the content of the action element? If so, when do you use one vs the other?

Reply:
Nevermind. my bad :)